libraries/chain_kv/include/b1/session/boost_memory_allocator.hpp

```diff
@@ -0,0 +1,79 @@
+#pragma once
+
+#include <memory>
+
+#include <boost/pool/pool.hpp>
+
+#include <b1/session/bytes_fwd_decl.hpp>
+
+namespace eosio::session {
+
+// \brief Defines a memory allocator backed by a boost::pool instance.
+//
+// \remarks This also demonstrates the "memory allocator" concept.  If we want to introduce another
```

Comment:
Is there a reason for introducing a custom allocator concept instead of using std::allocator?

Reply:
I have none.  Will make it conform to that concept.